### PR TITLE
Upgrade vim version to 9.0.0050 to fix CVEs: 2022-2257, 2022-2264, 2022-2284, 2022-2285, 2022-2286, 2022-2287

### PR DESCRIPTION
**Pull request by mandeepsplaha**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
This PR upgrades the vim version to fix several CVEs: 2022-2257, 2022-2264, 2022-2284, 2022-2285, 2022-2286, 2022-2287.

###### Change Log  

Upgraded the vim version from 8.2.5172 to 9.0.0050.

CVEs fixed by this change:
- 2022-2257
- 2022-2264
- 2022-2284
- 2022-2285
- 2022-2286
- 2022-2287

###### Does this affect the toolchain?  

**NO**

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2022-2257
- https://nvd.nist.gov/vuln/detail/CVE-2022-2264
- https://nvd.nist.gov/vuln/detail/CVE-2022-2284
- https://nvd.nist.gov/vuln/detail/CVE-2022-2285
- https://nvd.nist.gov/vuln/detail/CVE-2022-2286
- https://nvd.nist.gov/vuln/detail/CVE-2022-2287

###### Test Methodology

- Buddy Build: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=212576&view=results

### CODE REVIEW

File: SPECS/vim/vim.spec [hunk 1]
```diff
@@ -181,9 +178,6 @@ fi
 %{_datarootdir}/vim/vim*/colors/lists/default.vim
 %{_datarootdir}/vim/vim*/defaults.vim
 %{_datarootdir}/vim/vim*/filetype.vim
-%{_datarootdir}/vim/vim82/syntax/nosyntax.vim
-%{_datarootdir}/vim/vim82/syntax/syntax.vim
```
Comment (oliviacrain) [inline SPECS/vim/vim.spec:185]: These two files look like they're going to be moved to the `extras` subpackage. Can you confirm that the base package is still functional without these files after this change?

File: SPECS/vim/vim.spec [hunk 2]
```diff
@@ -181,9 +178,6 @@ fi
...
-%{_datarootdir}/vim/vim82/autoload/dist/ft.vim
```
Comment (oliviacrain) [inline SPECS/vim/vim.spec:186]: Same comment as above.

File: SPECS/vim/vim.spec [hunk 3]
```diff
@@ -125,6 +125,9 @@ fi
 %{_datarootdir}/vim/vim*/scripts.vim
 %{_datarootdir}/vim/vim*/spell/*
 %{_datarootdir}/vim/vim*/syntax/*
+%exclude %{_datarootdir}/vim/vim90/syntax/nosyntax.vim
+%exclude %{_datarootdir}/vim/vim*/syntax/syntax.vim
```
Comment (oliviacrain) [inline SPECS/vim/vim.spec:129]: Missed a `vim*` -> `vim90` conversion here!

**Review by oliviacrain — APPROVED** [review]
Looks good! Just one small comment.

**Review by oliviacrain — CHANGES_REQUESTED** [review]
Whoops, didn't see the accidental revert of my `mc`/`opensc`/`python3` changes. Will block until that's fixed.

File: SPECS/vim/vim.spec (hunk 1 above)
Comment (mandeepsplaha) [reply SPECS/vim/vim.spec:187]: Fixed this by adding the files back after updating their paths.

**Review by oliviacrain — APPROVED** [review]

**Pull request merged**